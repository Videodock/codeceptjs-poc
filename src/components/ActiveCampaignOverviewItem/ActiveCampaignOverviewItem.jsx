import React from 'react';
import * as PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/styles';
import { useTranslation } from 'react-i18next';
import { Typography } from '@material-ui/core';
import images from '../../theme/images';
import styles from './ActiveCampaignOverviewItem.styles';

const useStyles = makeStyles(styles);

export default function ActiveCampaignOverviewItem ({ scoreImage, scoreTitle, campaignTitle, campaignDeadline, onClick }) {
  const { t }   = useTranslation('campaign');
  const classes = useStyles();

  return (
    <div className={classes.root} onClick={onClick}>
      <div className={classes.imageContainer}>
        <img className={classes.image} src={scoreImage || images.scorePlaceholder} alt={t('overview.score')} />
      </div>
      <div className={classes.metaContainer}>
        <Typography className={classes.caption}>
          {scoreTitle}
        </Typography>
        <Typography className={classes.title} variant="h2">
          {campaignTitle}
        </Typography>
        <Typography>
          {campaignDeadline ? `${campaignDeadline} ${t('days_to_go')}` : null}
        </Typography>
      </div>
    </div>
  );
}

ActiveCampaignOverviewItem.propTypes = {
  scoreImage      : PropTypes.string,
  scoreTitle      : PropTypes.string,
  campaignTitle   : PropTypes.string,
  campaignDeadline: PropTypes.number,
  onClick         : PropTypes.func,
};
