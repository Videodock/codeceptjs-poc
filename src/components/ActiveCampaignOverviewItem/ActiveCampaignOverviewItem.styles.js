import { createStyles } from '@material-ui/styles';

export default ({ palette, spacing, typography }) => createStyles({
  root: {
    width          : 244,
    backgroundColor: palette.common.white,
  },
  imageContainer: {
    padding        : 5,
    textAlign      : 'center',
    backgroundColor: palette.border.lightGrey,
  },
  image: {
    width : 144,
    height: 203,
  },
  metaContainer: {
    padding: spacing(),
  },
  caption: {
    color        : palette.primary.main,
    fontFamily   : "Open Sans",
    fontSize     : typography.pxToRem(12),
    fontWeight   : 600,
    textTransform: 'uppercase',
    letterSpacing: 2,
  },
  title: {
    fontSize    : typography.pxToRem(21),
    marginBottom: 0,
  },
});
