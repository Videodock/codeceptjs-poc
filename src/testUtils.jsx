import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
