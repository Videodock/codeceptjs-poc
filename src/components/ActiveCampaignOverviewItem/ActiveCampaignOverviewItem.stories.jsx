import React from 'react';
import { storiesOf } from '@storybook/react';
import ActiveCampaignOverviewItem from './ActiveCampaignOverviewItem';

storiesOf('ActiveCampaignOverviewItem', module)
  .add('basic', () => <ActiveCampaignOverviewItem />);
