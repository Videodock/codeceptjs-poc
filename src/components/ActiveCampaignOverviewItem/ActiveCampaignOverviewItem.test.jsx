import React from 'react';
import { render } from '../../testUtils';
import ActiveCampaignOverviewItem from './ActiveCampaignOverviewItem';

const dummyProps = {
  scoreImage   : undefined,
  scoreTitle   : "bb",
  campaignTitle: "cc",
};

describe('<ActiveCampaignOverviewItem />', () => {
  it('renders and matches snapshot', () => {
    const { container } = render(<ActiveCampaignOverviewItem {...dummyProps} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('it renders the score title, campaign title and campaign deadline correctly', () => {
    const { getByText } = render(<ActiveCampaignOverviewItem {...dummyProps} />);

    expect(getByText('bb')).toBeTruthy();
    expect(getByText('cc')).toBeTruthy();
  });
});
