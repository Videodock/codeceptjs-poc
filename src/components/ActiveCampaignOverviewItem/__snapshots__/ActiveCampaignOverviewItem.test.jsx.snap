// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActiveCampaignOverviewItem /> renders and matches snapshot 1`] = `
<div
  class="makeStyles-root-1"
>
  <div
    class="makeStyles-imageContainer-2"
  >
    <img
      alt="Score"
      class="makeStyles-image-3"
      src="[object Object]"
    />
  </div>
  <div
    class="makeStyles-metaContainer-4"
  >
    <p
      class="MuiTypography-root makeStyles-caption-5 MuiTypography-body1"
    >
      bb
    </p>
    <h2
      class="MuiTypography-root makeStyles-title-6 MuiTypography-h2"
    >
      cc
    </h2>
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      dd
    </p>
  </div>
</div>
`;
